test-lib.sh

```diff
@@ -4,24 +4,46 @@
 #
 # Always use sourced from a specific container testfile
 #
-# reguires definition of CID_FILE_DIR
-# CID_FILE_DIR=$(mktemp --suffix=<container>_test_cidfiles -d)
-# reguires definition of TEST_LIST
-# TEST_LIST="\
-# ctest_container_creation
-# ctest_doc_content"
 
 # Container CI tests
 # abbreviated as "ct"
 
+# run ct_init before starting the actual testsuite
+
 # may be redefined in the specific container testfile
 EXPECTED_EXIT_CODE=0
-export TESTSUITE_RESULT=0
 
 # define UNSTABLE_TESTS if not already defined, as this variable
 # is not mandatory for containers
 UNSTABLE_TESTS="${UNSTABLE_TESTS:-""}"
 
+# following lines are added only for preserving backwards compatibility
+# and can be removed, when PR #306 is fully integrated to all container
+# repositories.
+TESTSUITE_RESULT=0
+APP_ID_FILE_DIR="/tmp/APP_ID_FILE_DIR_$RANDOM"
```

Comment:
Variable $RANDOM is specified where?

Reply:
from bash manpage:

```
RANDOM Each time this parameter is referenced, it expands to a random integer between
              0 and 32767.  Assigning a value to RANDOM initializes (seeds) the sequence  of
              random  numbers.  If RANDOM is unset, it loses its special properties, even if
              it is subsequently reset.
```

Reply:
Why random and not `mktemp -d`?

Reply:
This is because I did not really comprehend what sourcing a bash script means and I thought that it is not possible.
But meanwhile I learnt that it should be possible, so I will rewrite it to mktemp -d.

Reply:
Hm, that's the detail I missed -- doing something on a filesystem when the script is just sourced is not perfect. At minimum it should be always done only once (so maybe checking whether the variable is defined already and not reset it when the directory is created?).

However, reading the comment, as this is just a temporary solution until the new approach is integrated everywhere, let's not engineer it here :)

Reply:
> Hm, that's the detail I missed -- doing something on a filesystem when the script is just sourced is not perfect.

`mktemp` offers `--dry-run` option, which only prints the name of temporary directory, but does not create it. This option can be used here.